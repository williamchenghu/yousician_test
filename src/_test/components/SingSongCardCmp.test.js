import React from 'react'
import { songList } from '../_mocks_/songList'
import SingSongCardCmp from '../../components/SingleSongCardCmp'

describe('Container: SingSongCardCmp', () => {
  const props = {
    onChangeRating: jest.fn(),
    songDetails: songList[0],
  }
  describe('activate: SingSongCardCmp', () => {
    it('should be defined', () => {
      // the component should be define
      expect(SingSongCardCmp).toBeDefined()
    })

    it('should render correctly', () => {
      //prepare
      let wrapper = shallow(<SingSongCardCmp {...props} />)
      //check
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })

  describe('actions: SingSongCardCmp', () => {
    let wrapper

    beforeEach(() => {
      wrapper = shallow(<SingSongCardCmp {...props} />)
    })
    afterEach(() => {
      jest.clearAllMocks()
    })
    it('should called onChangeRating when the rating star has been changed', () => {
      //run the function before getDefaultJobId get resovled
      wrapper.find('#rating').simulate('change', songList[0], 5)
      //check
      expect(props.onChangeRating).toHaveBeenCalled()
    })
  })
})
