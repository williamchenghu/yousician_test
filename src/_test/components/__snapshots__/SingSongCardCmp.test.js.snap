// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container: SingSongCardCmp activate: SingSongCardCmp should render correctly 1`] = `
<div
  className="card padding-space"
>
  <div
    className="grid logo-border"
  >
    <img
      alt="Logo"
      className="logo"
      src="fingerprint-white.png"
    />
  </div>
  <div
    className="grid"
  >
    <LevelChartCmp
      level={13}
    />
  </div>
  <div
    className="grid"
  >
    <div>
      Lycanthropic Metamorphosis
    </div>
    <div
      className="card"
    >
      <div
        className="grid"
      >
        <RatingAPILayer
          direction="ltr"
          emptySymbol={
            <i
              className="material-icons"
            >
              star_border
            </i>
          }
          fractions={2}
          fullSymbol={
            <i
              className="material-icons"
            >
              star
            </i>
          }
          id="rating"
          initialRating={5}
          onChange={[Function]}
          onHover={[Function]}
          placeholderSymbol={
            Object {
              "backgroundColor": "red",
              "border": "5px double white",
              "borderRadius": "50%",
              "display": "inline-block",
              "height": 30,
              "width": 30,
            }
          }
          quiet={false}
          readonly={false}
          start={0}
          step={1}
          stop={5}
        />
      </div>
      <div
        className="grid"
      >
        The Yousicians
      </div>
    </div>
  </div>
</div>
`;
